docs/source/plugin/bot.rst

```diff
@@ -96,6 +96,122 @@ the :meth:`~sopel.bot.SopelWrapper.action` method::
     bot.action('does something')
 
 
+Do it with style
+================
+
+.. Custom role definition to apply custom style to inline text
+
+.. role:: red
+    :class: red
+
+.. role:: strongred
+    :class: strong red
+
+.. role:: strike
+    :class: strike
+
+.. role:: strongem
+    :class: strong em
+
+.. role:: spoiler
+    :class: spoiler
+
+
+When the bot talks, replies, or acts, it can do it with style: colors,
+**bold**, *italic*, _underline_, :strike:`striked`, or ``monospace``. IRC
+formatting works with control code, bytes you can use to tell IRC clients how
+to display some part of the text.
+
+.. seealso::
+
+    If you want to know more about `IRC formatting in general`__ and some of
+    its limitations, the modern IRC documentation may be of interest to you.
+
+    .. __: https://modern.ircdocs.horse/formatting.html
+
+However, dealing with control code yourself is not the most dev-friendly
+approach, hence the :mod:`sopel.formatting` module. It contains various
+functions to help you create stylized texts.
+
+Text styles
+-----------
+
+Let's dive into example, starting with :func:`~sopel.formatting.bold` text::
+
+    from sopel import formatting
+
+    bot.say(formatting.bold('This is some bold text!'))
+
+This will output a line like this:
+
+    [Sopel] **This is some bold text!**
```

Comment:
Here I wonder how these output examples will interact with syntax highlighting. Isn't this file using Python syntax by default for pre-blocks like this?

And also: Why is Sopel's nick in the examples wrapped by `[]` instead of `<>`? The square brackets make it look (to me) like the function is secretly wrapped in `@plugin.output_prefix('[Sopel] ')`. 😁

Reply:
I'll change that to ` <>` .

And this is not a code/pre block, because the previous line doesn't end with `::`, but a simple, unique, `:`. In that case, it's a blockquote element.